src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/NewtonsoftReferenceAnalyzer.cs

```diff
@@ -9,42 +9,70 @@
 
 namespace Microsoft.DotNet.UpgradeAssistant.Steps.Packages.Analyzers
 {
+    /// <summary>
+    /// Increases backward compatibility by using the Newtonsoft Serializer for ASP.NET Core.
+    /// </summary>
     public class NewtonsoftReferenceAnalyzer : IPackageReferencesAnalyzer
     {
         private const string NewtonsoftPackageName = "Microsoft.AspNetCore.Mvc.NewtonsoftJson";
 
         private readonly IPackageLoader _packageLoader;
         private readonly ILogger<NewtonsoftReferenceAnalyzer> _logger;
+        private readonly ITargetFrameworkMonikerComparer _tfmComparer;
 
         public string Name => "Newtonsoft.Json reference analyzer";
 
-        public NewtonsoftReferenceAnalyzer(IPackageLoader packageLoader, ILogger<NewtonsoftReferenceAnalyzer> logger)
+        public NewtonsoftReferenceAnalyzer(IPackageLoader packageLoader, ILogger<NewtonsoftReferenceAnalyzer> logger, ITargetFrameworkMonikerComparer tfmComparer)
         {
             _packageLoader = packageLoader ?? throw new ArgumentNullException(nameof(packageLoader));
             _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _tfmComparer = tfmComparer ?? throw new ArgumentNullException(nameof(tfmComparer));
         }
 
-        public async Task<PackageAnalysisState> AnalyzeAsync(IProject project, PackageAnalysisState state, CancellationToken token)
+        /// <summary>
+        /// Limits the step from being applied to the wrong project types.
+        /// </summary>
+        /// <param name="project">The project whose NuGet package references should be analyzed.</param>
+        /// <param name="token">The token used to gracefully cancel this request.</param>
+        /// <returns>True if the project is a web project targeting ASP.NET Core.</returns>
+        public Task<bool> IsApplicableAsync(IProject project, CancellationToken token)
         {
             if (project is null)
             {
                 throw new ArgumentNullException(nameof(project));
             }
 
-            if (state is null)
-            {
-                throw new ArgumentNullException(nameof(state));
-            }
-
             // This reference only needs added to ASP.NET Core exes
-            if (!(project.Components.HasFlag(ProjectComponents.AspNetCore) && project.OutputType == ProjectOutputType.Exe))
+            if (!(project.Components.HasFlag(ProjectComponents.AspNetCore)
+                && project.OutputType == ProjectOutputType.Exe
+                && !project.TargetFrameworks.Any(tfm => _tfmComparer.Compare(tfm, new TargetFrameworkMoniker("netcoreapp3.0")) < 0)))
```

Comment:
Technically, we'd want to ensure it's at least .net core 3.0, and not .net core 2.1. That being said, `.IsNetCore` is probably sufficient since 2.1 is not supported anymore